### PR TITLE
Add branch for TF muons to L1 ntuples

### PR DESCRIPTION
**Pull request by thomreis**
Add module to make L1 ntuples of track finder muons.
Can this be added as a cmsDriver customisation? Since not all TF have unpackers ready it would be most important with the simXmtfDigis products as inputs.

Activate uGMT autoBxRange by default. This will give less muons than before in the tests because only BX0 muons will be in the output of the uGMT. The output BX range is determined from the input with the smallest BX range, which is the OMTF in the tests with a range of only one BX.

### CODE REVIEW

**Comment by thomreis** [comment]
@rekovic I had a look at the size increase when producing the ntuple. Using the example cmsDriver options [1] and adding the TF ntuple the size increases by 4% (Tested with ~1600 events).

[1]
l1NtupleRECO -s RAW2DIGI --era=Run2_2016 --customise=L1Trigger/Configuration/customiseReEmul.L1TReEmulFromRAW --customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleAODEMU --customise=L1Trigger/Configuration/customiseUtils.L1TTurnOffUnpackStage2GtGmtAndCalo --conditions=auto:run2_data -n 200 --data --no_exec --no_output --filein=/store/data/Run2015D/DoubleEG/RAW-RECO/ZElectron-PromptReco-v4/000/260/627/00000/12455212-1E85-E511-8913-02163E014472.root --geometry=Extended2016,Extended2016Reco --customise=L1Trigger/Configuration/customiseReEmul.L1TEventSetupForHF1x1TPs

**Comment by rekovic** [comment]
@thomreis  Perfect. Thanks!

File: L1Trigger/L1TMuon/python/simGmtStage2Digis_cfi.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
     forwardTFInput = cms.InputTag("simEmtfDigis", "EMTF"),
     #triggerTowerInput = cms.InputTag("simGmtCaloSumDigis", "TriggerTower2x2s"),
     triggerTowerInput = cms.InputTag("simGmtCaloSumDigis", "TriggerTowerSums"),
-    autoBxRange = cms.bool(False), # if True the output BX range is calculated from the inputs and 'bxMin' and 'bxMax' are ignored
+    autoBxRange = cms.bool(True), # if True the output BX range is calculated from the inputs and 'bxMin' and 'bxMax' are ignored
```
Comment (rekovic) [inline L1Trigger/L1TMuon/python/simGmtStage2Digis_cfi.py:15]: @thomreis It looks like that due to this change we now lost lots of re-emulated muons for abs(eta) > 1.2
See https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideL1TOfflineDev#Development_Integration
and comparison plots for v24.2 vs v24.1
(This PR integrated is tagged as l1t-integration-v24.2)

**Comment by thomreis** [comment]
@rekovic This is expected since the automatic BX range detection will find that, since OMTF has a BX range of only 1 (That means they send only BX0 muons), it needs to set the uGMT output BX range to 1 too.
Since we have this problem with EMTF, that was discussed in the meeting yesterday, that they have a shift in BX of +2, all those muons will not be propagated through the uGMT anymore.

For the BX 0 muons we should see the same number though if you can check this easily.

You can set the value back to false until the EMTF issue is solved if you think it is better for the moment. At some point we should set it to true though.

**Comment by rekovic** [comment]
@thomreis OK. Thanks.  Will leave it as as (AutoBxRange = True).

**Pull request merged**